cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.realtime.sinks;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.templates.etl.api.PipelineConfigurer;
+import co.cask.cdap.templates.etl.api.Property;
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.config.ETLStage;
+import co.cask.cdap.templates.etl.api.realtime.RealtimeSink;
+import co.cask.cdap.templates.etl.api.realtime.SinkContext;
+import co.cask.cdap.templates.etl.common.Properties;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * Real-time sink for Streams
+ */
+public class StreamSink extends RealtimeSink<StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamSink.class);
+
+  @Override
+  public void configure(StageConfigurer configurer) {
+    configurer.setDescription("Real-time sink that outputs to the specified stream.");
+    configurer.addProperty(new Property(Properties.Stream.NAME,
+                                        "The name of the stream to output to . Must be a valid stream name. " +
+                                          "The stream will be created if it does not exist",
+                                        true));
+    configurer.addProperty(new Property(Properties.Stream.DATA_FIELD,
+                                        "Field name in the generic record that contains the data to be written to " +
+                                          "the specified stream. The data could be either in avro/binary format or " +
+                                          "stream event byte buffer format. If unspecified, 'data' key is used and " +
```

Comment:
not sure what this means.  I think we can enforce that the field is of type bytes or string since that's what's available in the StreamWriter api.  If they need to convert to bytes they can always use the projection transform